modules/vebal/voting-list.service.ts

```diff
@@ -0,0 +1,188 @@
+import { Interface, getAddress } from 'ethers/lib/utils';
+import { capitalize, flatten, chain } from 'lodash';
+import { setRequestScopedContextValue } from '../../modules/context/request-scoped-context';
+import { AllNetworkConfigs } from '../../modules/network/network-config';
+import { networkContext } from '../../modules/network/network-context.service';
+import multicall3Abi from '../../modules/pool/lib/staking/abi/Multicall3.json';
+import {
+    LiquidityGaugesInfo,
+    RootGaugesInfo,
+    gaugeSubgraphService,
+} from '../../modules/subgraphs/gauge-subgraph/gauge-subgraph.service';
+import { getContractAt } from '../../modules/web3/contract';
+import { NetworkConfig } from '../network/network-config-types';
+import veBalHelpersAbi from './abi/veBalHelpers.json';
+import { prisma } from '../../prisma/prisma-client';
+import { Chain as PrismaChain, PrismaVotingList } from '@prisma/client';
+import { Chain as SubgraphChain } from '../subgraphs/gauge-subgraph/generated/gauge-subgraph-types';
+
+type LiquidityGaugeInfo = LiquidityGaugesInfo[number];
+interface GaugeInfo extends LiquidityGaugeInfo {
+    chain: PrismaChain;
+}
+type GaugesInfo = GaugeInfo[];
+
+export class VotingListService {
+    async syncVotingList() {
+        const promises = Object.values(AllNetworkConfigs).map(async (network) => fetchGaugesInfo(network));
+        const gaugeInfoForAllNetworks = await Promise.all(promises);
+        const gaugesInfo = flatten(gaugeInfoForAllNetworks);
+
+        console.log('Total gauges', gaugesInfo.length);
+
+        const validGauges = await filterValidGauges(gaugesInfo);
+
+        console.log('Total valid gauges', validGauges.length);
+
+        // console.log('First gauge record: ', validGauges[0]);
+        return validGauges;
+    }
+
+    async reloadVotingList(newGauges: GaugesInfo) {
+        console.log('Inserting', newGauges);
+
+        await prisma.prismaVotingList.deleteMany();
+
+        await prisma.prismaVotingList.createMany({
+            data: newGauges.map((gauge) => ({
+                address: getAddress(gauge.id),
+                chain: gauge.chain,
+                isKilled: gauge.isKilled,
+                relativeWeightCap: gauge.relativeWeightCap,
+                // poolId: gauge.poolId,
+            })),
+        });
+    }
+
+    async getList(): Promise<PrismaVotingList[]> {
+        return await prisma.prismaVotingList.findMany();
+    }
+
+    /**
+     * This query illustrates the pool related data that we need for each gauge
```

Comment:
This is one of the biggest pieces to finish the overall design of the feature: 

In the `veBalVotingList` resolver, I need to join the rows from PrismaVotingList with the pool data (from `prismaPool` and `prismaTokens`). 

`getPoolsForVotingList` is querying the pool data that I need (note that I also need to exclude BPT tokens from the result).

Each gauge row has a `poolId` column to perform the join with the previous data. I need a simple an efficient way to do that in the resolver but I never worked with prisma before, so I will need your help. 

Given that we number of rows is not huge, we could denormalise this pool data storing in the PrismaVotingList table. That would be super fast and flexible. But, I guess that you will have a better suggestion.

Reply:
It's fine to do the join as you did below. Only thing I'm not sure is how we should handle gauges of pools which we don't have in the backend currently. There are these special type pools that are not supported but where there are hardcoded gauges https://github.com/balancer/frontend-v2/blob/master/public/data/hardcoded-gauges.json